core/src/test/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityInstanceFilterShould.kt

```diff
@@ -0,0 +1,175 @@
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.android.core.trackedentity
+
+import com.google.common.collect.Lists
+import com.google.common.truth.Truth
+import org.hisp.dhis.android.core.common.AssignedUserMode
+import org.hisp.dhis.android.core.common.BaseIdentifiableObject
+import org.hisp.dhis.android.core.common.DateFilterPeriod
+import org.hisp.dhis.android.core.common.DatePeriodType
+import org.hisp.dhis.android.core.common.FilterPeriod
+import org.hisp.dhis.android.core.common.ObjectShould
+import org.hisp.dhis.android.core.common.ObjectWithUid
+import org.hisp.dhis.android.core.common.RelativePeriod
+import org.hisp.dhis.android.core.enrollment.EnrollmentStatus
+import org.hisp.dhis.android.core.event.EventStatus
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnitMode
+import org.hisp.dhis.android.network.trackedEntityInstanceFilter.TrackedEntityInstanceFilterDTO
+import org.junit.Test
+import org.mockito.internal.util.collections.Sets
+import java.io.IOException
+import java.text.ParseException
+
+class TrackedEntityInstanceFilterShould :
+    TrackedEntityInstanceFilterCommonShould("trackedentity/tracked_entity_instance_filter.json"),
+    ObjectShould {
+    @Test
+    @Throws(IOException::class, ParseException::class)
+    override fun map_from_json_string() {
+        val trackedEntityInstanceFilterDTO = deserialize(TrackedEntityInstanceFilterDTO.serializer())
+        val trackedEntityInstanceFilter = trackedEntityInstanceFilterDTO.toDomain()
+
+        teiFilterCommonAsserts(trackedEntityInstanceFilter)
+        Truth.assertThat(trackedEntityInstanceFilter).isEqualTo(
+            teiFilterSample(),
+        )
+    }
+
+    companion object {
+        @Suppress("LongMethod")
+        private fun teiFilterSample(): TrackedEntityInstanceFilter {
+            return TrackedEntityInstanceFilter.builder()
+                .uid("klhzVgls081")
+                .code("assigned_none")
+                .name("Ongoing foci responses")
+                .displayName("Ongoing foci responses")
+                .created(BaseIdentifiableObject.parseDate("2019-09-27T00:19:06.590"))
+                .lastUpdated(BaseIdentifiableObject.parseDate("2019-09-27T00:19:06.590"))
+                .description("Foci response assigned to someone, and the enrollment is still active")
+                .sortOrder(2)
+                .program(ObjectWithUid.create("M3xtLkYBlKI"))
+                .eventFilters(
+                    Lists.newArrayList<TrackedEntityInstanceEventFilter>(
```

Comment:
Replace all the occurrences of `Lists.newArrayList<>` by `listOf` to remove dependency with `com.google.common.collect.Lists`